.jenkins/library/vars/helpers.groovy

```diff
@@ -149,6 +150,40 @@ def testSamplesLinux(boolean lvi_mitigation, String oe_package) {
     """   
 }
 
+/** Returns current Windows Intel SGX PSW version */
+def getPSWversion() {
+    return powershell(
+        label: "Intel SGX PSW version",
+        script: "(Get-Item C:\\Windows\\System32\\sgx_urts.dll).VersionInfo.ProductVersion",
+        returnStdout: true
+    ).trim()
+}
+
+/** Returns current Windows Intel SGX DCAP version */
+def getDCAPversion() {
+    return powershell(
+        label: "Intel SGX DCAP version",
+        script: "(Get-Item C:\\Windows\\System32\\sgx_dcap_ql.dll).VersionInfo.ProductVersion",
+        returnStdout: true
+    ).trim()
+}
+
+/** Checks to see if Windows Intel SGX PSW is installed correctly */
+def verifyPSWinstall() {
+    def installedVer = powershell(
+        label: "Verify Intel SGX PSW installation",
+        script: """
+            (Get-WmiObject Win32_PnPSignedDriver| select DeviceName, DriverVersion, Manufacturer | where {\$_.DeviceName -like "*Guard Extensions Platform*"}).DriverVersion
+        """,
+        returnStdout: true
+    ).trim()
+    def dllVer = getPSWversion()
+    if(installedVer != dllVer) {
+        print("[Error] Installed PSW version ${installedVer} does not match dll version ${dllVer}")
+    }
+    assert installedVer == dllVer
```

Comment:
What happens if this assert fails? Is it logged?

Reply:
The above statement `if(installedVer != dllVer) {` would catch and print out an error message if the assert were to fail.